views/web_views.py

```diff
@@ -275,6 +272,50 @@ def join_mailing_list():
 
     return jsonify(success=True, message=gettext("Thanks for signing up!"))
 
+@cross_origin()
+@app.route("/oeth-subscribe", methods=["POST"], strict_slashes=False)
+def oeth_subscribe():
```

Comment:
nit: Since only `source` difference between this function and the other one, why one extract this into a common function to reduce code duplication?